networkx/algorithms/centrality/current_flow_closeness.py

```diff
@@ -69,12 +76,14 @@ def current_flow_closeness_centrality(G, weight=None, dtype=float, solver="lu"):
     """
     if not nx.is_connected(G):
         raise nx.NetworkXError("Graph not connected.")
+    N = G.number_of_nodes()
+    if N < 3:
+        raise nx.NetworkXError("Graph has fewer than three vertices.")
```

Comment:
here, you can consider raising `NetworkXUnfeasible` instead of `NetworkXError`(ref. [docs](https://networkx.org/documentation/stable/reference/exceptions.html#networkx.NetworkXUnfeasible))
```suggestion
        raise nx.NetworkXUnfeasible("Graph has fewer than three vertices.")
```

Reply:
I'm not sure I agree that fewer than three vertices (side note: does NetworkX typically use nodes?) makes for an "instance without a feasible solution." I am open to being convinced though.

Reply:
I don't know the details of this `current_flow_closeness_centrality` algorithm. But, usually, the notion of closeness, centrality, or betweenness is not very meaningful when it comes to 1-node, 2-node, or 0-node graphs. Because for a 2-node graph, both nodes are equally important(or central) and in a 1-node graph it's just one node and in a 0-node graph there is nothing, so how can someone define centrality? In the paper, the author(s) decided not to consider such cases, and if we want to follow the paper strictly then it seems reasonable to me to say that such cases are unfeasible. But, I might be wrong or missing something. Why do you think these should not be considered as unfeasible cases?

Also, in networkx for the `betweenness_centrality` algorithm, the following are the outputs for 1-node, 2-node, and 0-node graphs:
```.py
>>> G = nx.Graph()
>>> nx.betweenness_centrality(G)
{}
>>> G.add_node(1)
>>> nx.betweenness_centrality(G)
{1: 0.0}
>>> G.add_node(2)
>>> nx.betweenness_centrality(G)
{1: 0.0, 2: 0.0}
>>> G.add_edge(1,2)
>>> nx.betweenness_centrality(G)
{1: 0.0, 2: 0.0}
```
So, you can either return something like this or you can strictly follow the paper and raise these exceptions. Currently, this is the output(Using the same graph as above):

```.py
>>> nx.current_flow_closeness_centrality(G)
{1: 1.0, 2: 1.0}
>>> G.remove_node(2)
>>> nx.current_flow_closeness_centrality(G)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
    return {ordering[node]: 1 / value for node, value in betweenness.items()}
                            ~~^~~~~~~
ZeroDivisionError: float division by zero
```

Reply:
I might be arguing semantics, but to me `NetworkXUnfeasible` exceptions should be reserved for things that are unexpected terminations of algorithms (as `NetworkXUnfeasible` inherits from `NetworkXAlgorithmError`), not particular cases where the solution is either trivial (if we do not strictly follow the paper) or not defined (if we do).

For the record, I think it makes sense intuitively to extend the algorithm to the case with two nodes by considering both nodes to be equally "central," and to raise a `NetworkXPointlessConcept` exception with appropriate error message for the null graph case, as discussed in another review comment). I'm unsure whether it makes sense to not error out for the single node empty graph case, unless we arbitrarily agree on a normalization value.

Reply:
> but to me NetworkXUnfeasible exceptions should be reserved for things that are unexpected terminations of algorithms (as NetworkXUnfeasible inherits from NetworkXAlgorithmError)

I think it makes sense to not use `NetworkXUnfeasible` here and I think my confusion came from some places where I've seen it being used where the solution was not defined and not unfeasible. Like, at some places in `dag.py`, `NetworkXUnfeasible` is raised when `G` is not a DAG and the algorithm is defined only for DAGs. And, I also used `NetworkXUnfeasible` in `panther_similarity` for when the `source` node is an isolated node(PR https://github.com/networkx/networkx/pull/7110). Thank you for pointing this out :)

> For the record, I think it makes sense intuitively to extend the algorithm to the case with two nodes by considering both nodes to be equally "central," and to raise a NetworkXPointlessConcept exception with appropriate error message for the null graph case, as discussed in another review comment). I'm unsure whether it makes sense to not error out for the single node empty graph case, unless we arbitrarily agree on a normalization value.

I think, currently, the output of the 2-node case is fine, and for 1-node and 0-node cases we can either return `{1: 1.0}` and `{}` respectively or appropriate error messages. Both seem equally reasonable to me. But, for 0-node case, we would be changing the API as now `NetworkXPointlessConcept` exception would be replaced by `{}`. So, we might need a deprecation there. And, for 0-node case it can be considered as fixing a buggy behavior I think. LMK what you think.

Reply:
> I think it makes sense to not use `NetworkXUnfeasible` here and I think my confusion came from some places where I've seen it being used where the solution was not defined and not unfeasible.

Take my interpretation with a grain of salt! I'm just sharing how I personally understand this. I wonder what @dschult thinks of this.

> I think, currently, the output of the 2-node case is fine, and for 1-node and 0-node cases we can either return `{1: 1.0}` and `{}` respectively or appropriate error messages. Both seem equally reasonable to me.

I would be in favor of raising errors - the [docs](https://networkx.org/documentation/stable/_modules/networkx/exception.html#AmbiguousSolution) for `AmbiguousSolution` say this best: "in the face of ambiguity, refuse the temptation to guess;" which is what we'd be doing by arbitrarily normalizing centrality to be 1 for a single node graph.

Similarly, computing centrality for the null graph is about as close as one can come to a textbook example of a `NetworkXPointlessConcept`.